block/fork.go

```diff
@@ -0,0 +1,181 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/dymensionxyz/dymint/node/events"
+	"github.com/dymensionxyz/dymint/types"
+	sequencers "github.com/dymensionxyz/dymint/types/pb/rollapp/sequencers/types"
+	uevent "github.com/dymensionxyz/dymint/utils/event"
+	"github.com/dymensionxyz/dymint/version"
+)
+
+const (
+	LoopInterval = 15 * time.Second
+)
+
+// MonitorForkUpdateLoop monitors the hub for fork updates in a loop
+func (m *Manager) MonitorForkUpdateLoop(ctx context.Context) error {
+	err := m.checkForkUpdate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(LoopInterval) // TODO make this configurable
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			if err := m.checkForkUpdate(ctx); err != nil {
+				continue
+			}
+		}
+	}
+}
+
+// checkForkUpdate checks if the hub has a fork update
+func (m *Manager) checkForkUpdate(ctx context.Context) error {
+	rollapp, err := m.SLClient.GetRollapp()
+	if err != nil {
+		return err
+	}
+
+	lastBlock, err := m.Store.LoadBlock(m.State.Height())
+	if err != nil {
+		return err
+	}
+
+	if m.shouldStopNode(rollapp, lastBlock) {
+		err = m.createInstruction(rollapp)
```

Comment:
will be called every `LoopInterval = 15 * time.Second` until manual reboot?
the freezeNode doesn't freeze this loop